### ISSUE TITLE
Not all organizations in list

### ISSUE THREAD

**Issue by ffes** [k0shk0sh/FastHub#271]
Great to have support for organizations, but not all the organizations I am part of are shown. Only the ones that I am owner of are in the list and not those that I am "just" member of.

In my list `cygwin` and `editorconfig` are not shown in the list but I am member of both of them.

**Comment by k0shk0sh** [comment]
You have to go to your auth apps in GitHub website and click on request org to allow FastHub to access them.

**Comment by k0shk0sh** [comment]
Please follow #249

**Comment by ffes** [comment]
I will try that out. AFK this weekend

I also have the ForkHub app installed on my phone and there is just works.

**Comment by k0shk0sh** [comment]
@ffes not really sure why tho,  could it be that you need to login again go authorize the app? Could you try that and let me know.

**Comment by ffes** [comment]
I authorized ForkHub only once. I see in my settings that ForkHub uses a Personal Access Token. I don't see **any** mention of FastHub on my settings page.

**Comment by k0shk0sh** [comment]
You'll definitely need to re-login for it to work, as the org scope wasn't in FastHub and its newly added. could you give it a shot?

**Comment by ffes** [comment]
I miss-read your reply, sorry.

I tried to re-login in FastHub, but it didn't help. Still only the see those 2 orgs.

After reading the issue you mentioned again, I saw that FastHub is in the "Authorized Applications" section. And there I can request access for the other two orgs. Didn't do that though. Seems that it needs to be handled by an admin of the org.

Still it is very confusing, especially since with ForkHub it just works! All orgs are shown and can be used to browse to any repo of that org that I'm allowed to see. But they use the Personal Access Token.

**Comment by k0shk0sh** [comment]
@ffes alright, I'll try and see what I can do about. This issue now becomes in my top list as I thought it was just a GitHub thingy that you need to do that manually

**Comment by christopherkeller** [comment]
@ffes @k0shk0sh New FastHub user and I also had the problem where it wasn't showing a couple of orgs. The issue seems to be that newer orgs are set to deny all / allow approved third party applications by default.  Once I went into the Authorized Applications section in my personal settings and explicitly granted FastHub access to the org (I'm the org admin), FastHub was able to see the org with out issue. Just to double check, I went back into the org settings and now FastHub appears in the list of approved third-party applications.

So, at least for me, it's working as intended. Hopefully that's helpful to someone.

**Comment by k0shk0sh** [comment]
@christopherkeller thanks for clarifying that, and yes indeed GitHub has changed this behavior and we can't do nothing about it.

**Comment by christopherkeller** [comment]
Older orgs allow all 3rd party apps just fine by default, but any orgs created more recently have the newer permissions style.  I was definitely confused at first as well, as it seemed odd that 3/5 orgs were showing up just fine but 2 weren't. Until I read the GitHub docs and then came across this thread. Then I realized the two invisible ones were created in the last year or two and that must have been the cutoff for the behavior change. 

At any rate, seems like a FAQ (if there is one) would alleviate the issue in the future for newcomers. A good Android app was sorely lacking, thanks for taking the time to work on this.

**Comment by k0shk0sh** [comment]
@christopherkeller I'll try and find a place for your comment to be placed in. People rarely search before submitting or reacting to an issue.

**Comment by ffes** [comment]
But still it remains strange that with the same settings in all the orgs I am part of, all orgs are shown in ForkHub and only two in FastHub.

**Comment by ffes** [comment]
Unfortunately that is very hard for me to test ATM.

There is another problem (I will open a new issue for this), I use Firefox for Android as my default browser and I can't get the authorization to work with that.

I can try later today to change some settings and see how I can use Chrome as the default browser.

**Comment by k0shk0sh** [comment]
@ffes  any news on this? its really still bugging me & want to confirm on the behaviour before I take any action.

**Comment by ffes** [comment]
Using the browser to authorize doesn't change anything. But then again, FastHub still shows up as an Authorized Application. So I don't see any difference between using the builtin form or the external browser, expect that the browser is much more of a hassle.

So I tried to create a personal access token manually on https://github.com/settings/tokens While creating I checked `gist`, `repo` and `user` (just like ForkHub). With that token I asked the API for my orgs and all of them showed up.

```bash
curl -vvv -H "Authorization: Bearer MY_SECRET_TOKEN" https://api.github.com/user/orgs
```

**Comment by k0shk0sh** [comment]
@ffes Alright, I guess I got this working fine, so you can no longer see FastHub in your Autho Apps however FastHub will use AccessToken to behave like the app you mentioned.

**Comment by cprogrammer1994** [comment]
I am not able to list my organizations. It keeps saying i have no organizations.

**Comment by k0shk0sh** [comment]
@cprogrammer1994 you might need to follow whatever discussion here or you could wait for tonight release which will fix this for good

**Comment by k0shk0sh** [comment]
@cprogrammer1994  || @ffes  anyone of you guys is available right now to test this out?

**Comment by k0shk0sh** [comment]
@ffes Please check your Email. I'm awaiting for the news :p

**Comment by ffes** [comment]
I have the mail. I will test it later this afternoon.

**Comment by ffes** [comment]
I get this error when I try to login. Don't if it is important for this, but I have two-factor enabled

![screenshot_20170414-150241](https://cloud.githubusercontent.com/assets/3225656/25043996/4249f46a-2124-11e7-89ef-5f29f3becd24.png)

**Comment by k0shk0sh** [comment]
@ffes Oops, didn't check against that at all. 
I'll send u new build once I'm home and fix this 😥😥

**Comment by k0shk0sh** [comment]
@ffes Please do check your email again :) and sorry for all the trouble caused.

**Comment by k0shk0sh** [comment]
So after debugging the whole day on this,

Using access tokens: 
- bros:
  - you have control over **FastHub** permissions.
- cons:
  - using 2fa will log you out every few seconds due to the lifetime of the otp code.

conclusion: 

as mentioned in [here](https://developer.github.com/v3/auth/#working-with-two-factor-authentication) orgs that have restrictions enabled, the user required to send them request to enable an application to access them.

I'm going to stick with the current implementation due to the fact that using 2fa has short time of lifetime before github returns an unauthorizes access.

Therefore, this issue going to be flagged as close until there is a proper away to suites 2fa lifetime.

**Comment by ffes** [comment]
I just tested the apk you sent me and all the orgs are shown. When I tried to post my findings I ran into the authorization problems you described. So I understand your decision.

**Comment by k0shk0sh** [comment]
@ffes I really hope you do, it would be bad experience for you, me & others lol.

**Thanks for your understanding**

**Comment by k0shk0sh** [comment]
I guess this would be helpful to avoid confusion again.

![Imgur](http://i.imgur.com/GpQARYc.png)

**Comment by k0shk0sh** [comment]
@ffes access token login has been implemented now, you could login using a pre-generated access token.

**Comment by ffes** [comment]
@k0shk0sh I just tried the new access token access in v2.0.1. A bit cumbersome, but **it works**. All my orgs show up and everything seems to work so far.

**Comment by k0shk0sh** [comment]
@ffes glad to hear that, however if you are facing some errors, like can't close this and open that or permission is denied, please do update your access token permissions :)
